code/mob/living/critter/small_animal.dm

```diff
@@ -447,22 +456,113 @@ ABSTRACT_TYPE(/mob/living/critter/small_animal)
 		if (prob(10))
 			src.audible_message("[src] purrs!",\
 			"You purr!")
+		if (src.ai.enabled && ispug(user) && prob(10))
+			ON_COOLDOWN(src, "recent_pug_pet", 15 SECONDS)
+			src.ai.priority_tasks += src.ai.get_instance(/datum/aiTask/sequence/goalbased/critter/attack, list(src, src.ai.default_task))
+			src.ai.interrupt()
+			src.visible_message("<span class='notice'>[src] recoils and hisses at [user]'s attempt to pet them, then goes for the jugular!</span>")
+			playsound(src.loc, 'sound/voice/animal/cat_hiss.ogg', 50, 1)
+
+	Crossed(atom/movable/M as mob)
+		..()
+		if (!isalive(src))
+			return
+		if(ishuman(M))
+			var/mob/living/carbon/human/H = M
+			if(prob(10) && src.is_annoying)
+				src.visible_message("<span class='combat'>[src] weaves around [H]'s legs and trips [him_or_her(H)]!</span>")
+				H.setStatus("resting", duration = INFINITE_STATUS)
+				H.force_laydown_standup()
+			else if (prob(4) && ishuman(M))
+				boutput(src, "<span class='notice'>You weave around [H] to [pick("show your affection!", "get them to feed you.", "annoy them for no reason in particular.")]</span>")
+				boutput(H, "<span class='notice'>[src] weaves around you, waving their tail around. A bunch of hair clings to your clothes and some gets in your nose.</span>")
+				H.emote("sneeze")
+
+	seek_target(range)
+		. = list()
+		for (var/atom/movable/AM in hearers(range, src))
```

Comment:
```suggestion
		for (var/mob/living/AM in hearers(range, src))
```
we don't need all movable atoms i n range surely?

Reply:
We do need objects to locate living tails though, as they are objects. So either i could look for movables, or i could run hearers twice?

Reply:
Do hearers once, store the result, then do two filtered loops - one for mob/living and the other for /obj/tail. It should be reasonably efficient that way.

Reply:
Sounds a lot more efficient, you're right.